article/2022/requirements-txt-vs-setup-py-in-python.md

```diff
@@ -90,47 +90,47 @@ install_requires =
 
-Now if you need to use both notations (pretty much always to be honest), then you need to make sure that you won’t repeat yourself.
+现在，如果你需要同时使用这两种表方式（老实说几乎都是如此），那么你需要确保不会重复。
 
-In case you are using both, your **`setup.py` (and/or `setup.cfg`) files should include the list of abstract dependencies, whereas the `requirements.txt` file must contain the concrete dependencies** with the specific pins for each package version (using `==` pin).
+如果你同时使用这两种, 你的 **`setup.py` (和/或者 `setup.cfg`) 文件需要包含抽象依赖项列表**，而`requirements.txt` **文件必须包含使用不同方式(例如 `==`) 来指定限制包版本的确切依赖项**。
 
-Whereas `install_requires` (i.e. `setup.py`) defines the dependencies for a single project, [Requirements Files](https://pip.pypa.io/en/latest/user_guide/#requirements-files) are often used to define the requirements for a complete Python environment.
+尽管 `install_requires`（例如 `setup.py`）定义了单个项目的依赖关系，而 [`requirements.txt`]((https://pip.pypa.io/en/latest/user_guide/#requirements-files)) 通常用于定义完整 Python 环境的依赖关系。
 
-Whereas `install_requires` requirements are minimal, requirements files often contain an exhaustive listing of pinned versions for the purpose of achieving [repeatable installations](https://pip.pypa.io/en/latest/user_guide/#repeatability) of a complete environment.
+尽管 `install_requires` 要求很少，但 `requirements.txt` 通常需要包含详尽的固定版本列表，以实现完整环境的[可重复安装](https://pip.pypa.io/en/latest/user_guide/#repeatability)
 
-- [Python Documentation](https://packaging.python.org/en/latest/discussions/install-requires-vs-requirements/#requirements-files)
+- [Python 文档](https://packaging.python.org/en/latest/discussions/install-requires-vs-requirements/#requirements-files)
 
-## Final Thoughts
+## 总结
 
-In today’s tutorial, we discussed the importance of proper dependency management when it comes to developing Python projects and applications. We discussed the purpose of `requirements.txt` file and how it can be used alongside setup files of `setuptools` (i.e. `setup.py` and `setup.cfg`) in order to ensure that other developers can install, run, develop or even test the source code of a Python package.
+在这篇文章中,我们介绍了在开发 Python 项目和应用程序时应用适当的依赖管理的重要性。 也介绍了 `requirements.txt` 文件的用户和如何将它与 `setuptools` (例如 `setup.py` 和 `setup.cfg`)的设置文件一起使用，以确保其他开发者可以安装，运行，开发，甚至测试 Python 包的源代码。
 
-As already highlighted, `setuptools` are not exactly a replacement for `requirements.txt` files. In most cases, you will need both files to be present in order to properly manage your package dependencies in a way that they can also be re-distributed later on.
+正如强调的那样, `setuptools` 不完全是 `requirements.txt` 文件的替代品. 在大多数情况下, 你可能需要同时使用这两个文件，来正确的管理你的包依赖项和便于对他们进行重新发布。
 
-If you are wondering now how you can potentially distribute your Python package and make it available on PyPI so that it can also be installed through the `pip` package manager, make sure to read the article below.
+如果你想知道如何发布你的 Python 包并使其在 PyPI 上可用，除此之外还可以通过 `pip` 包管理器来安装它，请关注接下来的文章。
```

Comment:
除此之外还可以通过 `pip` 包管理器来安装它
=>
以便可以通过 `pip` 包管理器来安装它

将 so that 翻译为因果关系更合理
```suggestion
如果你想知道如何发布你的 Python 包并使其在 PyPI 上可用，以便可以通过 `pip` 包管理器来安装它，请关注接下来的文章。
```